### ISSUE TITLE
[DPO/KTO] Mixtral Load Balancing Loss

### ISSUE THREAD

**Issue by claralp** [huggingface/trl#1544]
Question towards @lewtun @kashif:

We noticed that the load balancing loss (aux_loss) that is implemented in MoEs [modeling_mixtral.py#L1244](https://github.com/huggingface/transformers/blob/main/src/transformers/models/mixtral/modeling_mixtral.py#L1391) is not added to the loss implemented in DPO/KTO trainers.  

Isn't this needed so that the load balancing between experts is still guaranteed after fine-tuning or does the KL loss penalization sufficiently prevents the router weights from changing too much?

**Comment by PhilipMay** [comment]
> my initial intuition was that we are changing the weight of the model and we keep the routing of the experts in-tact.

So your suggestion is to use a lora / qlora adapter that does not target the router layers?

**Comment by claralp** [comment]
@kashif, so I guess it is important to set `output_router_logits=False` for preference tuning.  
However, it looks like setting this does not disable the router weights from being trained.  
So where do you disable the weights for routing of experts in preference tuning?  
If that is a required manual step, maybe a note should be added to the trl MoE examples.

**Comment by kashif** [comment]
on clarification... in the preference tuning losses we are not explicitly back-proping over the categorical cross entropy loss of the models but rather via the log-probs obtained from the logits... so in this case these methods are not doing anything to the router's weights in MoE models

**Comment by claralp** [comment]
@kashif we could observe a gradient in the gate weights even when backpropagating via the log-probs obtained from the logits (as in DPO/KTO).  
If the gradient calculation is not explicitly switched off for the 'gate' weights or if they are excluded from the `target_modules` in the PEFT config (default), these gating weights seem to be trained by DPO/KTO
